e2e/internal/keyexec/keyexec.go

```diff
@@ -0,0 +1,361 @@
+func RunKeyCmd(t *testing.T, cmdPath string, commands []string, file, stdin string) (string, []byte, error) {
+	argv := []string{"key"}
+	argv = append(argv, commands...)
+
+	if file != "" {
+		argv = append(argv, file)
+	}
+
+	cmd := fmt.Sprintf("%s %s", cmdPath, strings.Join(argv, " "))
+	execKey := exec.Command(cmdPath, argv...)
+
+	stdinRun, err := ioutil.TempFile("", "")
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
+	defer stdinRun.Close()
+
+	_, err = io.WriteString(stdinRun, stdin)
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
+
+	_, err = stdinRun.Seek(0, os.SEEK_SET)
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
+
+	execKey.Stdin = stdinRun
+	out, err := execKey.CombinedOutput()
+
+	return cmd, out, err
+}
+
+// QuickTestKey will export a private, and public key (0), and then import them. This is used after
+// generating a newpair.
+func QuickTestKey(t *testing.T) {
+	e2e.LoadEnv(t, &testenv)
+
+	tmpTestDir := filepath.Join(testenv.TestDir, "quick_test_key_verify")
+
+	tests := []struct {
+		name    string
+		private bool
+		armor   bool
+		succeed bool
+	}{
+		{
+			name:    "quick test public",
+			private: false,
+			armor:   false,
+			succeed: true,
+		},
+		{
+			name:    "quick test public  armor",
+			private: false,
+			armor:   true,
+			succeed: true,
+		},
+		{
+			name:    "quick test private",
+			private: true,
+			armor:   false,
+			succeed: true,
+		},
+		{
+			name:    "quick test private armor",
+			private: true,
+			armor:   true,
+			succeed: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, test.WithoutPrivilege(func(t *testing.T) {
+			os.RemoveAll(tmpTestDir)
+			os.MkdirAll(tmpTestDir, os.ModePerm)
+			var c string
+			var b []byte
+			var err error
+
+			if tt.private {
+				c, b, err = ExportPrivateKey(t, filepath.Join(tmpTestDir, "export_key.asc"), "0\n", tt.armor)
+			} else {
+				c, b, err = RunKeyCmd(t, testenv.CmdPath, []string{"export"}, filepath.Join(tmpTestDir, "export_key.asc"), "0\n")
+			}
+			if tt.succeed {
+				if err != nil {
+					t.Log("Command that failed: ", c)
+					t.Log(string(b))
+					t.Fatalf("unepexted failure: %v", err)
+				}
+				if tt.private {
+					t.Run("remove_private_keyring_before_importing", test.WithoutPrivilege(func(t *testing.T) { RemoveSecretKeyring(t) }))
+				} else {
+					t.Run("remove_public_keyring_before_importing", test.WithoutPrivilege(func(t *testing.T) { RemovePublicKeyring(t) }))
+				}
+				t.Run("import_private_keyring_from", test.WithoutPrivilege(func(t *testing.T) {
+					c, b, err := ImportPrivateKey(t, filepath.Join(tmpTestDir, "export_key.asc"))
+					if err != nil {
+						t.Log("command that failed: ", c, string(b))
+						t.Fatalf("Unable to import key: %v", err)
+					}
+				}))
+			} else {
+				if err == nil {
+					t.Log(string(b))
+					t.Fatalf("unexpected succees running: %v", c)
+				}
+			}
+		}))
+	}
+}
+
+func KeyNewPair(t *testing.T, user, email, note, psk1, psk2 string, push bool) (string, []byte, error) {
```

Comment:
add comments on all functions before the definition, so that it can be clear what they're doing :+1: